doc/glossary.md

```diff
@@ -0,0 +1,313 @@
+
+#### Darjeeling
+
+A work-in-progress integrated OpenTitan top level.
+
+#### DD
+
+Digital Design.
+
+#### DV
+
+Design Verification.
+
+#### DVsim
+
+Design verification simulator tool.
+
+#### E2E
+
+End-to-End.
+
+#### Earl Grey
+
+The first discrete OpenTitan top level.
+See [Earl Grey](../hw/top_earlgrey/README.md) for a the full specification.
+
+#### EDA
+
+Electronic Design Automation (tool).
+
+#### EDN
+
+Entropy Distribution Network.
+
+#### ePIC
+
+Embedded Position Independent Code.
+
+#### ePMP
+
+Enhanced Physical Memory Protection.
+
+A RISC-V extension to the physical memory protection feature.
+Also known as "Smepmp".
+
+#### FIDO2
+
+Fast Identity Online version 2.
+
+A collection of protocol specifications for authenticating a user with a device or server.
+
+#### FIPS
+
+Federal Information Process Standards.
+
+#### Flash
+
+Non-volatile reprogrammable memory.
+
+#### FPGA
+
+Field Programmable Gate Array.
+
+#### FPV
+
+Formal Property Verification.
+
+#### GF2
+
+Galois Field of order 2.
+
+#### GPIO
+
+General-Purpose Input/Output.
+
+#### HDL
+
+Hardware Description Language.
+
+#### HMAC
+
+Hash-based Message Authentication Code.
+
+#### HSM
+
+Hardware Security Module.
+
+#### HyperDebug
+
+Custom development board for testing and debugging OpenTitan devices.
+
+#### I<sup>2</sup>C
+
+Inter-Integrated Circuit.
+
+#### Ibex
+
+The 32-bit RISC-V core used in OpenTitan chips.
+
+#### IP
+
+Intellectual Property.
+
+#### KMAC
+
+Keccak Message Authentication Code.
+
+#### LC
+
+Life-Cycle (controller).
+
+#### Mask ROM
+
+Read-Only Memory encoded into the "mask" used to fabricate silicon.
+
+#### MCU
+
+Microcontroller Unit.
+
+#### Mux
+
+Multiplexer.
+
+#### Netlist
+
+A list of connections between nodes of a circuit.
+
+#### OTBN
+
+OpenTitan Big Number - a RISC-V-like big number accelerator core.
+
+#### OTP
+
+One-Time Programmable memory.
+
+#### PIC
+
+Position-Independent Code.
+
+#### Pinmux
+
+Pin Multiplexer.
+
+### PKI
+
+Public Key Infrastructure.
+
+#### PMP
+
+Physical Memory Protection.
+
+### PoR
+
+Power-on-Reset.
+
+#### PQC
+
+Post-Quantum Cryptography.
+
+#### PRNG
+
+Pseudorandom Number Generator.
+
+#### PWM
+
+Pulse-Width Modulation.
+
+#### RAM
+
+Random-Access Memory.
+
+#### RFC
+
+Request For Comments.
+
+#### RISC-V
+
+The open source instruction set architecture used for OpenTitan cores.
+
+#### ROM
+
+Read-Only Memory.
+
+#### RoT
+
+Root of Trust.
+
+#### RTL
+
+Register-Transfer Level.
+
+#### RV DM
+
+RISC-V Debug Module.
+
+#### SHA
+
+Secure Hash Algorithms.
+
+#### Silicon
+
+The metal semiconductor material that most chips are made from.
+
+#### Smoketest
+
+A simple test used to check that some feature works to some degree, even if not in depth.
+
+#### SoC
+
+System on a Chip.
+
+#### SPI
+
+Serial Peripheral Interface.
+
+#### SRAM
+
+Static Random-Access Memory.
+
+#### Stapping
+
+Configuration of GPIO input settings used to change the state of a chip.
```

Comment:
Yeah, this was a hard one.

How about:

> GPIO signals sampled by OpenTitan for configuration.

Reply:
I'll use this, but FYI some lifecycle states sample straps continuously so they're not only for early boot.

PS @HU90m, our version of mdbook [has this feature](https://github.com/rust-lang/mdBook/blob/master/CHANGELOG.md#mdbook-0430), but are you happy for me to use it style-wise?

```markdown
#### Configuration Straps / Pin Straps {#straps}
```

This sets the ID of the header so that links can be written shorter. I'm not adding links in this PR, so will leave it out for now. I'm not sure how GitHub handles these.

Reply:
We should probably come up with a fuller definition in time, as straps don't seem to be well-defined elsewhere in the repo. In general, they exist to be a simple way to configure the system, that is typically static (pull up/down with a resistor) but less frequently is dynamic. They are one of the only ways to provide early-boot configuration (though only a small number of bits of config), but the pin/connection can then often be used/reused for other things e.g. GPIO.

Reply:
Sure, happy to defer this. I've taken your definition for now anyway.